projects/packages/connection/src/class-connection-notice.php

```diff
@@ -147,35 +147,54 @@ function ( $connected_admin ) {
 			echo '</form>';
 			?>
 			<script type="text/javascript">
-				jQuery( document ).ready( function( $ ) {
-					$( '#jp-switch-connection-owner' ).on( 'submit', function( e ) {
-						var formData = $( this ).serialize();
-						var submitBtn = document.getElementById( 'jp-switch-connection-owner-submit' );
-						var results = document.getElementById( 'jp-switch-user-results' );
+				( function() {
+					const switchOwnerButton = document.getElementById('jp-switch-connection-owner');
+					if ( ! switchOwnerButton ) {
+						return;
+					}
 
+					switchOwnerButton.addEventListener( 'submit', function ( e ) {
+						e.preventDefault();
+
+						const submitBtn = document.getElementById('jp-switch-connection-owner-submit');
 						submitBtn.disabled = true;
 
-						$.ajax( {
-							type        : "POST",
-							url         : "<?php echo esc_url( get_rest_url() . 'jetpack/v4/connection/owner' ); ?>",
-							data        : formData,
-							headers     : {
-								'X-WP-Nonce': "<?php echo esc_js( wp_create_nonce( 'wp_rest' ) ); ?>",
-							},
-							success: function() {
-								results.innerHTML = "<?php esc_html_e( 'Success!', 'jetpack-connection' ); ?>";
-								setTimeout( function() {
-									$( '#jetpack-notice-switch-connection-owner' ).hide( 'slow' );
-								}, 1000 );
+						const results = document.getElementById('jp-switch-user-results');
+						results.innerHTML = '';
+						results.classList.remove( 'error-message' );
+
+						fetch(
+							<?php echo wp_json_encode( esc_url_raw( get_rest_url() . 'jetpack/v4/connection/owner' ), JSON_HEX_TAG | JSON_HEX_AMP ); ?>,
+							{
+								method: 'POST',
+								headers: {
+									'X-WP-Nonce': <?php echo wp_json_encode( wp_create_nonce( 'wp_rest' ), JSON_HEX_TAG | JSON_HEX_AMP ); ?>,
+								},
+								body: new URLSearchParams( new FormData( this ) ),
 							}
-						} ).done( function() {
-							submitBtn.disabled = false;
-						} );
-
-						e.preventDefault();
-						return false;
-					} );
-				} );
+						)
+							.then( response => response.json() )
+							.then( data => {
+								submitBtn.disabled = false;
```

Comment:
(feel free to ignore this comment)

I wonder if the button should only be re-enabled when the submission fails. If it succeeded we probably don't want them clicking it a second time because we're going to remove the form in a second anyway...

Yeah, the old code re-enabled it too, but maybe that was a bug.

Reply:
Also fixed in https://github.com/Automattic/jetpack/pull/29087/commits/455661b4b156df53b2c2bebcb86438e395deba91